content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,235 @@
+
+In Step 2.a we kept  our application on the heartbeat of the XRPL by periodically polling it for updates. In JavaScript,
+our client is connecting to the XRPL via [WebSockets](https://en.wikipedia.org/wiki/WebSocket). Our client has a 
+permanent bidirectional connection to the XRPL, which allows us to subscribe to events that the server sends out. This 
+saves resources on the server, which now only sends out data we explicitly asked for when a change happens, as well as 
+the client which does not have to sort through incoming data for relevant changes. This also reduces the complexity of the 
+application and saves us a couple of lines of code.
+
+To get the application running at this stage of development, run the following command:
+
+```console
+npm run async-subscribe
+```
+
+### 3. Display an Account
+
+**Full code for this step:** [`3_account.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/3_account.js).
+
+Now that we have a  permanent connection to the XRPL and some code to bring the delivered data to life on our 
+screen, it's time to add some "wallet" functionality by managing am individual account. For this step, we will momentarily 
+hardcode the address of the account, and save the complexities of key handling for later. The application, after finishing 
+this step should look like this:
+
+![Screenshot: Step 3, show account information](img/javascript-wallet-3.png)
+
+To get the application running at this stage of development, run the following command:
+
+```console
+npm run account
+```
+
+### 4. Show Account's Transactions
+
+**Full code for this step:** [`4_tx-history.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/4_tx-history.js).
+
+At this point, your wallet shows the account's balance getting updated, but doesn't give you any clue about how this state
+came about, namely the actual transactions that caused the updates. So, our next step is to display the account's 
+transaction history and to keep it up to date:
+
+To get the application running at this stage of development, run the following command:
+
+![Screenshot: Step 4, show transaction history](img/javascript-wallet-4.png)
+
+```console
+npm run tx-history
+```
+
+
+### 5. Send XRP
+
+**Full code for this step:** [`5_send-xrp.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/2_send-xrp.js).
+
+Up until now we have enabled our app to query and display data from the XRPL. Now it's time to actively participate in
+the ledger by enabling out application to send transactions. For now, you can stick to sending direct XRP payments 
+because there are more complexities involved in sending issued tokens. 
+
+But before that, it is time to take that wallflowery appearance of our application and give it a more usable and 
+nicer looking UI. 
```

Comment:
We're doing both in the first step, maybe this should be split into two steps considering they're both significant changes to the app? 

Or maybe instead of saying `before that` we should say "In addition to that, we'll improve the UI in this step so it's easier to understand what's going on.

Reply:
It's a separate step now